contribs/gmf/src/datasource/Manager.js

```diff
@@ -685,88 +724,119 @@ const exports = class {
    * @private
    */
-  handleDataSourceFilterRulesChange_(dataSource, value) {
-
-    // Skip data sources that are not filtrables OR those that do not have
-    // the WMS ogcType, i.e. those that do not have an OpenLayers layer
-    // to update
-    if (dataSource.filtrable !== true ||
-        dataSource.ogcType !== ngeoDatasourceOGC.Type.WMS
-    ) {
-      return;
-    }
-
+  getDataSourceLayer_(dataSource) {
+    dataSource = /** @type {!gmf.DataSource} */ (dataSource);
     const id = olBase.getUid(dataSource.gmfLayer);
     const item = this.treeCtrlCache_[id];
-    googAsserts.assert(item);
+    if (item == undefined) {
+      return;
+    }
     const treeCtrl = item.treeCtrl;
+    return gmfLayertreeSyncLayertreeMap.getLayer(treeCtrl);
+  }
 
-    const layer = gmfLayertreeSyncLayertreeMap.getLayer(treeCtrl);
+  /**
+   * Update layer filter parameter according to data sources filter rules
+   * and dimensions filters.
+   * @param {ol.layer.Base} layer The layer to update.
+   * @private
+   */
+  updateLayerFilter_(layer) {
     googAsserts.assert(
       layer instanceof olLayerImage ||
       layer instanceof olLayerTile
     );
 
     const source = layer.getSource();
-    googAsserts.assert(
+    if (!(
       source instanceof olSourceImageWMS ||
-      source instanceof olSourceTileWMS
-    );
+      source instanceof olSourceTileWMS)) {
+      return;
+    }
 
-    const filtrableLayerName = dataSource.getFiltrableOGCLayerName();
-    const projCode = treeCtrl.map.getView().getProjection().getCode();
-    const filterString = dataSource.visible ?
-      this.ngeoRuleHelper_.createFilterString({
-        dataSource,
-        projCode
-      }) :
-      null;
+    const params = source.getParams();
+    const layersParam = params['LAYERS'];
+    const layersList = layersParam.split(',');
+    googAsserts.assert(layersList.length >= 1);
 
     const filterParam = 'FILTER';
-    let filterParamValue = null;
-
-    if (filterString) {
-      const params = source.getParams();
-      const layersParam = params['LAYERS'];
-      const layersList = layersParam.split(',');
-      googAsserts.assert(layersList.length >= 1);
-
-      if (layersList.length === 1) {
-        // When there's only one layer in the `LAYERS` parameters, then
-        // the filter string is given as-is.
-        filterParamValue = filterString;
-      } else {
-        // When there's more then one layer, then each filter must be wrapped
-        // between parenthesis and the order must also match the `LAYERS`
-        // parameter as well.
-        const filterParamValues = [];
-        for (let i = 0, ii = layersList.length; i < ii; i++) {
-          if (layersList[i] === filtrableLayerName) {
-            filterParamValues.push(`(${filterString})`);
-          } else {
-            filterParamValues.push('()');
+    const filterParamValues = [];
+    let hasFilter = false;
+    for (let i = 0, ii = layersList.length; i < ii; i++) {
```

Comment:
```
layersList.foreach((layer) {
...
})
```
?

Reply:
I would prefer `for (const layer of layersList) {`
Easier to step into with the debugger;